main/command/src/main/scala/sbt/BasicCommands.scala

```diff
@@ -27,7 +27,10 @@ object BasicCommands {
 
   def helpParser(s: State) =
     {
-      val h = (Help.empty /: s.definedCommands)(_ ++ _.help(s))
+      val h = (Help.empty /: s.definedCommands) { (a, b) ⇒
+        a ++
+          (try b.help(s) catch { case ex: Throwable ⇒ Help.empty })
```

Comment:
yeah, maybe use `NonFatal(e)` so you don't catch OOM/Interrupted errors that would have to be handled differently.